open_instruct/ppo_vllm_thread_ray_gtrl_weka.py

```diff
@@ -0,0 +1,1780 @@
+            )
+            worker_policy = ray_process_cls.options(
+                num_cpus=self.num_cpus_per_actor,
+                num_gpus=self.num_gpus_per_actor,
+                scheduling_strategy=scheduling_strategy,
+            ).remote(world_size, rank, 0, master_addr, master_port)
+            self.models.append(worker_policy)
+
+
+def main(args: Args, dataset_config: DatasetConfig, model_config: ModelConfig):
+    calculate_runtime_args(args, model_config)
+
+    # set up experiment tracking and seeds
+    all_configs = {}
+    beaker_config = None
+    if is_beaker_job():
+        args.checkpoint_output_dir = os.environ.get("CHECKPOINT_OUTPUT_DIR", None)
+        beaker_config = maybe_get_beaker_config()
+        all_configs.update(vars(beaker_config))
+    all_configs.update(**asdict(args), **asdict(dataset_config), **asdict(model_config))
+    if args.with_tracking:
+        import wandb
+
+        wandb.init(
+            project=args.wandb_project_name,
+            entity=args.wandb_entity,
+            sync_tensorboard=True,
+            config=all_configs,
+            name=args.run_name,
+            save_code=True,
+            tags=[args.exp_name] + get_wandb_tags(),
+        )
+    writer = SummaryWriter(f"runs/{args.run_name}")
+    writer.add_text(
+        "hyperparameters",
+        "|param|value|\n|-|-|\n%s" % ("\n".join([f"|{key}|{value}|" for key, value in vars(args).items()])),
+    )
+
+    # create a tokenizer (pad from right)
+    config = AutoConfig.from_pretrained(model_config.model_name_or_path, revision=model_config.model_revision)
+    tokenizer = AutoTokenizer.from_pretrained(
+        model_config.model_name_or_path, revision=model_config.model_revision, padding_side="right"
+    )
+    if config.architectures == "LlamaForCausalLM" and config.bos_token_id == 128000:
+        tokenizer.pad_token_id = 128002  # <|reserved_special_token_0|>
+    else:
+        tokenizer.add_special_tokens({"pad_token": "[PAD]"})  # NOTE: we do not resize the embedding
+    tokenizer.chat_template = CHAT_TEMPLATES[dataset_config.chat_template]
+
+    # create the dataset
+    dataset_dict = DatasetDict()
+    dataset_processor = SFTGroundTruthDatasetProcessor(tokenizer=tokenizer, config=dataset_config)
+    if len(args.dataset_train_splits) != len(args.dataset_mixer_dict) and len(args.dataset_train_splits) == 1:
+        args.dataset_train_splits = [args.dataset_train_splits[0]] * len(args.dataset_mixer_dict)
+        print(f"Dataset splits not provided for all datasets. Using the same {args.dataset_train_splits[0]} split for all datasets.")
+    if len(args.dataset_eval_splits) != len(args.dataset_eval_mixer_dict) and len(args.dataset_eval_splits) == 1:
+        args.dataset_eval_splits = [args.dataset_eval_splits[0]] * len(args.dataset_eval_mixer_dict)
+        print(f"Dataset splits not provided for all datasets. Using the same {args.dataset_eval_splits[0]} split for all datasets.")
```

Comment:
Btw @hamishivi this appears to be a bug in other implementations: if I mix dataset without specifying multiple train splits, the mix is prob not successful

Reply:
good catch. This probably affected me, although it doesnt quite explain why just MATH data alone isnt working well :(